src/main/java/org/hpccsystems/jdbcdriver/HPCCConnection.java

```diff
@@ -72,25 +76,42 @@ public class HPCCConnection implements Connection
     private String               catalog = HPCCJDBCUtils.HPCCCATALOGNAME;
 
     private Platform                    hpccPlatform              = null;
-    private HPCCWsClient                wsClient                  = null;
     private HPCCWsSQLClient             wsSQLClient               = null;
-    private String                      serverAddress;
-    private int                         wsEclPort;
+
     private String                      targetcluster;
     private String                      queryset;
     private String                      userName;
     private int                         pageSize;
     private int                         pageOffset;
     private int                         connectTimeoutMillis;
     private int                         readTimoutMillis;
-    private boolean                     hasTargetWsSQLBeenReached = false;
     private int                         eclResultLimit;
+    private boolean                     hasTargetWsSQLBeenReached = false;
 
     public HPCCConnection(Properties props)
     {
         this.connectionProps = props;
 
-        this.serverAddress = props.getProperty("ServerAddress", HPCCDriver.SERVERADDRESSDEFAULT);
+        String wsECLWatchAddress = props.getProperty("WsECLWatchAddress", HPCCDriver.SERVERADDRESSDEFAULT);
+        String wsSQLAddress = props.getProperty("ServerAddress", HPCCDriver.SERVERADDRESSDEFAULT);
+
+        if (wsSQLAddress.isEmpty())
+        {
+            SQLWarning warn = new SQLWarning("ServerAddress (WsSQL) not provided!");
```

Comment:
Should this be a warning if the connection will be aborted?

Reply:
This is a pre-established JDBC mechanism to communicate problems with the caller. 
It might have made more sense to label the method getNotifications rather than getWarnings:
https://docs.oracle.com/javase/8/docs/api/java/sql/SQLWarning.html